src/events/SDL_categories.c

```diff
@@ -0,0 +1,256 @@
+/*
+  Simple DirectMedia Layer
+  Copyright (C) 1997-2024 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+#include "SDL_internal.h"
+
+/* SDL event categories */
+
+#include "SDL_events_c.h"
+#include "SDL_categories_c.h"
+
+SDL_EventCategory SDL_GetEventCategory(Uint32 type) {
+    if (type >= SDL_EVENT_USER && type <= SDL_EVENT_LAST) {
+        return SDL_EVENTCATEGORY_USER;
+    }
+    switch (type) {
```

Comment:
You should use `if (type >= X && type <= Y)` for most of these, so we don't have to update this every time we add an event. Check the event documentation.

Reply:
Only `SDL_EVENT_WINDOW_*` and `SDL_DISPLAY_*` have a `_FIRST` and `_LAST`,
so the number of removed cases is limited to those.